dartagnan/src/main/java/com/dat3m/dartagnan/program/Thread.java

```diff
@@ -13,6 +13,8 @@ public class Thread extends Function {
     // Optional fields
     // Scope hierarchy of the thread
     private final Optional<ScopeHierarchy> optScopeHierarchy;
+
+    private final boolean hasScope;
```

Comment:
You don't need to add an extra field, you can do use ` !optScopeHierarchy.isEmpty()`